### ISSUE TITLE
Add averaging methods for classification metrics(Precion, Recall and F-score), in tensorflow.keras or tensorflow-addons

### ISSUE THREAD

**Issue by naveen-marthala** [keras-team/keras#15808]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**

- TensorFlow version (you are using): 2.4.3
- Are you willing to contribute it (Yes/No): Yes, but I have very little knowledge on tensorflow's graph execution works. So, I won't be able to contribute to be able to pass all checks during CI/CD.

**Describe the feature and the current behavior/state**.

Currently the [Recall](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/Recall)/[Precision ](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/Precision)metric available in tensorflow only provides binary averaged score; and [F1Score](https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/F1Score) and [FBetaScore](https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/FBetaScore). It would be very convenient and avoids the need to write (half-baked, at least in my case) custom metrics from user side, if other averaging methods available in sklearn, that is 'macro', 'micro' and 'samples'(for multi-label problems), are also implemented in either tensorflow or tensorflow-addons.

**Will this change the current api? How?**
Yes. by adding a new argument to also specify the averaging method user needs.

**Who will benefit from this feature?**
every one working on binary, multi-class, imbalanced classification and multi-label classification problems will benefit from this new feature.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Other info.**
All aforementioned metrics from sklearn's API docs are hyperlinked below: 
[sklearn's Recall](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html#sklearn.metrics.recall_score), [sklearn's Precision](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html#sklearn.metrics.precision_score), [sklearn's F1Score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score) and [sklearn's FBeta Score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.fbeta_score.html#sklearn.metrics.fbeta_score)

**Comment by sachinprasadhs** [comment]
From Tensorflow 2.13, new metrics are added under tf.keras.metrics.

The details for the newly added can be found in the links below.

1. [tf.keras.metrics.F1Score](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/F1Score)
2. [tf.keras.metrics.FBetaScore](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/FBetaScore)
3. [tf.keras.metrics.R2Score](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/R2Score)

Note that, since Tensorflow 2.13 is not yet released, this will be available only in Nightly version. 

Please let us know if this addresses your concern. Thanks

**Comment by naveen-marthala** [comment]
@sachinprasadhs , yes it addresses my concern.

**Comment by sachinprasadhs** [comment]
Thanks for confirming, Could you please close the issue as well.
